### ISSUE TITLE
TestAppendLineParser.test_get_recent_much fails on Windows

### ISSUE THREAD

**Issue by The-Compiler** [The-Compiler/qutebrowser#1565]
(already working on fixing this, just opening an issue so I have a place to write down things)

Since 44f6261 `TestAppendLineParser.test_get_recent_much` started to fail on AppVeyor:

```python
    def test_get_recent_much(self, lineparser):
        """Test get_recent with much data."""
        size = 64
        new_data = ['new data {}'.format(i) for i in range(size)]
        lineparser.new_data = new_data
        lineparser.save()
        data = '\n'.join(self.BASE_DATA + new_data)
        data = [e + '\n' for e in data[-(size - 1):].splitlines()]
        recent = lineparser.get_recent(size)
>       assert recent == data
E       assert ['ew data 59\...ew data 63\n'] == [' 58\n', 'new...ew data 63\n']
E         At index 0 diff: 'ew data 59\n' != ' 58\n'
E         Right contains more items, first extra item: 'new data 63\n'
E         Full diff:
E         + [' 58\n',
E         - ['ew data 59\n',
E         ? ^
E         +  'new data 59\n',
E         ? ^ +
E         'new data 60\n',
E         'new data 61\n',
E         'new data 62\n',
E         'new data 63\n']
```

Seems like we get a few bytes less than expected?

`data` (what we expect):

```python
[' 58\n',
 'new data 59\n',
 'new data 60\n',
 'new data 61\n',
 'new data 62\n',
 'new data 63\n']
```

`recent` (what we get):

```python
['ew data 59\n',
 'new data 60\n',
 'new data 61\n',
 'new data 62\n',
 'new data 63\n']
```

The `get_recent` implementation:

```python
    def get_recent(self, count=4096):
        with self._open('r') as f:
            f.seek(0, os.SEEK_END)
            size = f.tell()
            print("size {}".format(size))
            try:
                if size - count > 0:
                    offset = size - count
                else:
                    offset = 0
                f.seek(offset)
                data = f.readlines()
            finally:
                f.seek(0, os.SEEK_END)
        return data
```

The file contents at the start of `get_recent` (with another `self._open('r')`) look good:

```
[...]
new data 57
new data 58
new data 59
new data 60
new data 61
new data 62
new data 63
```

The size printed is `846`.

**Comment by The-Compiler** [comment]
This is something about CRLF.

This (expected) is 64 bytes:

```
 58\n (4)
new data 59\n (12)
new data 60\n (12)
new data 61\n (12)
new data 62\n (12)
new data 63\n (12)
```

And this (actual, with `\r` added by hand) is too:

```
ew data 59\r\n (12)
new data 60\r\n (13)
new data 61\r\n (13)
new data 62\r\n (13)
new data 63\r\n (13)
```

**Comment by The-Compiler** [comment]
Fixed via d6926f06227616c0f659ff1f39ec84f0e9b67465
